ChangeFeedLabSolution/ChangeFeedProcessor.cs

```diff
@@ -27,49 +30,49 @@ public static class ChangeFeedProcessor
+
+            return new EventHubProducerClient(eventHubNamespaceConnection, EventHubName);
+        }
         /// <summary>
         /// Processes modified records from Cosmos DB Collection into the Event Hub.
         /// </summary>
         /// <param name="documents"> Modified records from Cosmos DB collections. </param>
         /// <param name="log"> Outputs modified records to Event Hub. </param>
         [FunctionName("ChangeFeedProcessor")]
-        public static void Run(
+        public static async Task Run(
             //change database name below if different than specified in the lab
             [CosmosDBTrigger(databaseName: "changefeedlabdatabase",
             //change the collection name below if different than specified in the lab
             collectionName: "changefeedlabcollection",
             ConnectionStringSetting = "DBconnection",
             LeaseConnectionStringSetting = "DBconnection",
             LeaseCollectionName = "leases",
-            CreateLeaseCollectionIfNotExists = true)]IReadOnlyList<Document> documents, TraceWriter log)
+            CreateLeaseCollectionIfNotExists = true)]IReadOnlyList<Document> documents, ILogger log)
         {
-            // Create variable to hold connection string to enable event hub namespace access.
-#pragma warning disable CS0618 // Type or member is obsolete
-            string eventHubNamespaceConnection = ConfigurationSettings.AppSettings["EventHubNamespaceConnection"];
-#pragma warning restore CS0618 // Type or member is obsolete
-
-            // Build connection string to access event hub within event hub namespace.
-            EventHubsConnectionStringBuilder eventHubConnectionStringBuilder =
-                new EventHubsConnectionStringBuilder(eventHubNamespaceConnection)
-                {
-                    EntityPath = EventHubName
-                };
-
-            // Create event hub client to send change feed events to event hub.
-            EventHubClient eventHubClient = EventHubClient.CreateFromConnectionString(eventHubConnectionStringBuilder.ToString());
-
+            using EventDataBatch eventBatch = await producer.CreateBatchAsync();
             // Iterate through modified documents from change feed.
             foreach (var doc in documents)
             {
-                // Convert documents to json.
-                string json = JsonConvert.SerializeObject(doc);
+                // Convert documents to Json.
+                string json = JsonSerializer.Serialize(doc);
                 EventData data = new EventData(Encoding.UTF8.GetBytes(json));
-
-                // Use Event Hub client to send the change events to event hub.
-                eventHubClient.SendAsync(data);
+                if (!eventBatch.TryAdd(data))
+                {
+                    throw new Exception($"The event at { doc } could not be added.");

```

Comment:
This is going to spew the entire JSON blob back in the error message.   Is there a way that we can distill that into something more readable?   Maybe reporting the index into the collection where things failed?

Reply:
I'm investigating. It may take a while.

Reply:
It has been updated.